src/sqlfluff/rules/ambiguous/AM08.py

```diff
@@ -0,0 +1,95 @@
+"""Implementation of Rule AM08."""
+
+from typing import Optional, Tuple
+
+from sqlfluff.core.parser import (
+    ComparisonOperatorSegment,
+    KeywordSegment,
+    LiteralSegment,
+    WhitespaceSegment,
+)
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+
+
+class Rule_AM08(BaseRule):
+    """Join clauses should be present.
+
+    **Anti-pattern**
+
+    A join is used without specifying join condition.
+    This can cause unintended cross join behavior.
+    Prefer explicit const condition instead.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        JOIN baz;
+
+    **Best practice**
+
+    Use const condition instead.
+
+    .. code-block:: sql
+       :force:
+
+        SELECT
+            foo
+        FROM bar
+        JOIN baz
+        ON 1=1;
```

Comment:
Maybe it's just me, but I don't know if I like this as best practice. As per #5213, having an equality statement that's always true (whether `1=1`, `a=a` or `TABLE_B.COLUMN = TABLE_B.COLUMN`) - feels more like an antipattern to me. Especially if there are many tables involved. e.g.

```sql
select *
from a
join b
join c
join d
where b.x = c.x
and d.x = a.x
and 1=1
```
Technically this query would pass this rule, but I don't think it's any clearer whether the cross join was intentional or not.

..._however_ I do also understand the desire to give an option to dialects which don't support an explicit `CROSS` join 🤔 . I have two other suggestions:
1. We disable this rule be default for any dialects which don't support an explicit `CROSS JOIN`. Similar to how `AM02` is only enabled for dialects which support `UNION DISTINCT`.
2. We say that best practice in dialects which don't have a `CROSS JOIN` is to explicitly disable the rule to signal that it's intentional, e.g.
   ```sql
   select *
   from a
   join b  -- noqa: AM08
   ```

Do either of those options (or any other options you can think of) feel like a better solution to this problem?

Reply:
... interestingly, if we said that an "always true" condition like `1=1`, `a=a` or `TABLE_B.COLUMN = TABLE_B.COLUMN` doesn't count, then we wouldn't need separate rules for _implicit_ and _explicit_ cross joins, and you could resolve the original issue with one rule. 😄 

I'm still open to having two different rules for implicit/explicit, but could be a neat solution.

Reply:
@alanmcruickshank 

Please note that as of today, this fix only involves JOIN statements, not select from many targets, as in `SELECT * FROM a, b, c`.

In principle, I'd prefer to keep the default behavior to _refrain_ from emitting warnings instead of forcing users to inline disable.

I'd like to suggest additional test cases and then we can have another iteration on reviewing this.